gse_account/models/account.py

```diff
@@ -0,0 +1,56 @@
+# -*- coding: utf-8 -*-
+from odoo import fields, models, _
+from odoo.tools import (get_lang)
+
+
+class AccountMove(models.Model):
+    _inherit = "account.move"
+
+    custom_msg = fields.Boolean(string='Custom Message')
+
+    def action_invoice_sent(self):
```

Comment:
@iambenkis @MaganaAsiati 
 I see that we're essentially reusing much of the base function from Odoo's core code. Could we explore ways to minimize this to ensure our code is easier to maintain and better adheres to the DRY (Don't Repeat Yourself) principle? Ideally, we should aim to only override or extend the parts of the function that absolutely need customization, rather than duplicating existing logic. This approach would make our custom modifications more concise and straightforward to manage in future updates.

Looking forward to your thoughts on this.

Reply:
@iambenkis  the same function from the code base.
addons > account > model > account_move.py

Reply:
@iambenkis 
Have you tried to retrieve the custom or default email templated based on the company settings in a distinct method `_get_email_template` ?